### PR TITLE
Update DG for DeleteTags and some minor changes 

### LINKED ISSUE #254: Add DeleteSpecialisation/MedicalCondition/Pescriptions to DG's Implementation Section

### LINKED ISSUE #258: Developer Guide: Incorrect DeleteXYZ's Diagram and Text
image
image

### PR DESCRIPTION
**Pull request by kimshitong**
Closes #258 
Closes #254

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -401,6 +400,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+
+* `LogicManager#execute()` —  Executes the given user String input and returns a 'CommandResult'
+
+These operations are exposed in the `Ui` interface as `Ui#executeCommand()`.
+
+Given below is an example usage scenario and how the add `Specialisation` mechanism behaves at each step.
+
+Step 1. The user launches the application. The `Database` will be initialized with all data in the order that it was stored in.
+
+Step 2. The user inputs `delete_tag_d 2 Orthopaedic`  to delete an appointment's specialisation into MediConnect.
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:417]: `delete an appointment's` change to `delete a doctor's`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -401,6 +400,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+The `delete_tag_d` command calls `DeleteSpecialisationtCommandParser#parse` which parses the index argument which is the index of the doctor to delete  
+A new `DeleteSpecialisationtCommand` instance will be created
+
+Step 3. The created `DeleteSpecialisationtCommand` instance is returned to `LogicManager` and its `execute` method is called.
+`DeleteAppointmentCommand#execute` then calls `Model#getFilteredDoctorList` and retrieve the doctor with the given `Index`.
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:422]: `DeleteAppointmentCommand` change to `DeleteSpecialisationCommand`

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -401,6 +400,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+
+Step 2. The user inputs `delete_tag_d 2 Orthopaedic`  to delete an appointment's specialisation into MediConnect.
+The `delete_tag_d` command calls `DeleteSpecialisationtCommandParser#parse` which parses the index argument which is the index of the doctor to delete  
+A new `DeleteSpecialisationtCommand` instance will be created
+
+Step 3. The created `DeleteSpecialisationtCommand` instance is returned to `LogicManager` and its `execute` method is called.
+`DeleteAppointmentCommand#execute` then calls `Model#getFilteredDoctorList` and retrieve the doctor with the given `Index`. 
+Then, the specialisation will be removed from the doctor if exists and replace the existing doctor in Model with the command of `Model#setDoctor`.
+
+The example usage scenario for the delete patient and delete doctor mechanisms would be very similar to the above scenario.
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:425]: `delete patient and doctor` change to `delete prescription and medical condition`

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -401,6 +400,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+
+The following sequence diagram shows how the delete appointment operation would work and will be similar to medical condition and prescription:
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:427]: `delete appointment` change to `delete specialisation`

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -401,6 +400,36 @@ The following sequence diagram shows how the delete patient operation would work
...
+![DeleteSpecialisationSequenceDiagram](images/DeleteSpecialisationSequenceDiagram.png)
+
+The following activity diagram summarizes what happens when a user wants to delete an appointment/patient/doctor:
```
Comment (yanghengtang) [inline docs/DeveloperGuide.md:430]: `appointment/patient/doctor` change to `prescription/specialisation/medical condition`

File: docs/diagrams/DeleteXYZTagActivityDiagram.puml [hunk 6]
```diff
@@ -0,0 +1,28 @@
+@startuml
+'https://plantuml.com/activity-diagram-beta
+
+start
+:User enters DeleteXYZTagCommand with a specified index and parameter;
```
Comment (yanghengtang) [inline docs/diagrams/DeleteXYZTagActivityDiagram.puml:5]: if we define XYZ to be 'Specialisation, Medical Condition or Prescriptions', then there should be no 'Tag'

**Review by yanghengtang — COMMENTED** [review]
update the following changes

**Comment by kimshitong** [comment]
Thanks @yanghengtang for pointing out. Fixed.

**Review by yanghengtang — APPROVED** [review]

**Pull request merged**